### PR TITLE
Dropdown: onRenderTitle typing should be IRenderFunction<IDropdownOption[]> only

### LINKED ISSUE #8263: Dropdown: Should remove IDropdownOption from onRenderTitle's parameter type
### Environment Information 
 
- __Package version(s)__: 6.118.1 
- __Browser and OS versions__: All

#### Actual behavior:
 
onRenderTitle in Dropdown component can't handle IDropdownOption well.
It currently only accepts IDropdownOption[] as input type since taking the output of _getAllSelectedOptions().
 
#### Expected behavior:
If it only accepts array now,  we should remove IDropdownOption from onRenderTitle's input type.

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? Yes
 
Requested priority: Normal
 
Products/sites affected: (if applicable)

### PR DESCRIPTION
**Pull request by KevinTCoughlin**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #8263
- [x] Include a change request file using `$ npm run change`

#### Description of changes

This fixes the typing of `Dropdown`'s `onRenderTitle` to be `IRenderFunction<IDropdownOption[]>` instead of a union of `IRenderFunction<IDropdownOption | IDropdownOption[]>` introduced in https://github.com/OfficeDev/office-ui-fabric-react/pull/2386.

I believe this fix can be classified as a "bug fix" even though it is modifying the public API signature. This is because based on my grepping ([Github search](https://github.com/OfficeDev/office-ui-fabric-react/search?q=onRenderTitle&unscoped_q=onRenderTitle)), the `Dropdown` never passes a non-Array value to `onRenderTitle` since the above linked refactor.

I've checked OneDrive / SharePoint usage of `onRenderTitle` and all call-sites expect `IDropdownOption[]`.

I've also renamed `item` to `items` in the base implementation of `onRenderTitle` for clarity.

#### Focus areas to test

- CI should pass
- Tests should pass
- Dropdowns should function as they did previously

cc: @yenchanghsieh

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8270)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx [hunk 1]
```diff
@@ -401,10 +401,10 @@ export class DropdownBase extends BaseComponent<IDropdownInternalProps, IDropdow
   }
 
   /** Render text in dropdown input */
-  private _onRenderTitle = (item: IDropdownOption[]): JSX.Element => {
+  private _onRenderTitle = (items: IDropdownOption[]): JSX.Element => {
```
Comment (JasonGore) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx]: This technically isn't a match for the prop type and I'd hope TS would error on calling `map` on this arg.. I wonder if TS3.x would have caught this...

Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx]: It'll match with this change tho :)

Agreed that it is strange it wasn't caught by the compiler.

**Comment by KevinTCoughlin** [comment]
> In fact, I believe it would cause an exception since `map` is invoked on it. That justifies this change, I think.

I agree, and given that would cause a runtime exception i.e. a bug in the dependent code I think this is safe to merge as one of our "bug fix" typing changes.

If folks have problems matching the type send them my direction.

**Comment by KevinTCoughlin** [comment]
Waking up CLA bot

**Pull request merged**